### DISCUSSION TITLE
Synedit with Direct draw

### DISCUSSION THREAD

**Discussion by JaFi-cz** [pyscripter/SynEdit#61] [General]
Hello
I just make basic test - implement pyscripter/Synedit into PSpad. What I found the page draw is incredible slow in compare to old Synedit drawing.
Each Down or PageIp/Down "takes ages". I press PgDown and near 0,5s takes page appears.
I don't know what did I wrong...

**Comment by pyscripter** [comment]
This is very much against my experience and the experience of others.  See [this](https://en.delphipraxis.net/topic/6277-synedit-just-got-a-major-uplift/?tab=comments#comment-55407) for instance.

However the following can be causes of slowdown:
- Doing much stuff on OnPaintTransient
- Intendation guides

Many of the reasons for doing stuff using the OnPaintTransient do not exist any more (built-in bracket highlighting, syntax-checking using indicators etc.) and should be avoided if possible.  Mixing GDI with DirectDraw can slow things down.

What is the hardware/Windows version you are using?
Could you please do some testing to see what causes the slowness?  It may well be possible to address specific performance issues.
Please also note that DirectDraw is slow while debugging.

**Comment by JaFi-cz** [comment]
I need to draw matching code not only matching brackets (begin-end e.t.c.). 
No syntax. But direct draw was several times slower than old one. I was very surprised. I spent lot of time to modify my code to new SynEdit version but had to rollback it back.  
I will make some tests.

**Reply by pyscripter** [reply]
> I need to draw matching code not only matching brackets (begin-end e.t.c.).

You can also do that with indicators.   Much faster.  (See how bracket matching is done in SynEdit for a guide).
Please, try to disable all OnPaintTransint stuff and indicators and see whether it makes a difference.   Then enable them one by  one.   We need to find what is causing the slowness.

Scrolling is super fast and smooth here, so there must be a reason for the slowness.

**Reply by JaFi-cz** [reply]
OK, I will remove all painting and test speed. If it will be at least same speed as old painting, I will to study indicators usage and rewrite all using indicators.

**Reply by pyscripter** [reply]
I am anxiously waiting to hear the outcome :)

**Reply by pyscripter** [reply]
@JaFi-cz Any news on this?

**Reply by JaFi-cz** [reply]
Sorry, I am stuck with my job work. I have no time for my hobby projects now  :-(

**Reply by pyscripter** [reply]
Sure, making a living comes first!
However, it is easy to do a quick test.  Start a new project add a Synedit with a file and highlighter of your choice and see how smoothly and fast it scrolls.  If you avoid paint handlers and use indicators instead, you will get a similar speed in your editor app.
